lib/msal-node/src/crypto/PkceGenerator.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { PkceCodes } from '@azure/msal-common';
+import { PKCEConstants } from './../utils/NodeConstants';
+import { Base64Encode } from './../encode/Base64Encode';
+const crypto = require('crypto');
+
+/**
+ * https://tools.ietf.org/html/rfc7636#page-8
+ */
+export class PkceGenerator {
```

Comment:
same question about common since we do auth and pcke in browser

Reply:
This cannot be, as node.crypto is not available in browsers. These need to stay separate.